### ISSUE TITLE
Testing a new image

### ISSUE THREAD

**Issue by maybefreedom** [ignaciorlando/red-lesion-detection#2]
Hello!
I've read your code + article and find this project extremely interesing.
I think this might be obvious but I have trained my datasets and everything, now I need to test a new fundus image to see if the MA are detected. Is there a testing method I've overlooked?

Thank you and sorry again if the answer is obvious.

**Comment by ignaciorlando** [comment]
Hi @maybefreedom!

Thanks for your interest in our paper and code.

Yes, you can use this function:
`[red_lesion_segmentation, score_map] = full_red_lesion_segmentation(I, mask, L0, step, L, K, px, cnn_for_feature_extraction, vessel_segmentation, detector)`

This parameters are:
- `I`: the input image.
- `mask`: the FOV mask (you can use `mask = get_fov_mask(I, threshold)` to get it, try with different thresholds or use the values we reported in the paper).
- `L0`: the lowest value of L (you can set it in 3 as in our paper).
- `step`: the step to generate the L scales (you can set it in 3 as in our paper).
- `L`: the highest value of L (you can set it in 60 as in our paper).
- `K`: the maximum number of candidates (you can set it in 120 as in our paper).
- `px`: structures with less than `px` pixels will be removed (you can set it in 5 as in our paper).
- `cnn_for_feature_extraction`: the CNN structure to retrieve the features (you can use the CNN that you trained or load one of our pretrained networks).
- `vessel_segmentation`: the vessel segmentation. You can compute it using our segmentation method, or use a logical matrix with the same size of the image but full of zeros to ignore the segmentation feature.
- `detector`: the RF classifier, trained using our hybrid approach. You can use the model you learned or the pre-trained classifier we provide.

Let me know if you were able to run it!

Cheers,

Nacho
